app/modules/FrontModule/Registration/components/RegistrationForm/RegistrationForm.php

```diff
@@ -59,16 +59,16 @@ public function createComponentRegistrationForm(): Form
         ];
 
         $form->addRadioList('arrival', 'přijedu:', $time);
-        $invoice = $form->addCheckbox('invoice', 'chci fakturu na firmu');
+        $invoice = $form->addCheckbox('finvoice', 'chci fakturu na firmu');
```

Comment:
proc to "f"?

Reply:
Když projdeš třídu registration, tak to uvidíš. Máš tam k tomu i komentáře. Jde o to, že checkbox vrací bool a invoice v entitě, kam to chceš dostat je string. No a metoda, kterou jsi chtěl přidat (setMappedType) plní tyto property přímo z formulářových prvků, tedy se snažila do property invoice v entitě, která je datového typu string přiřadit bool z formulářového prvku. 
Takže jsem vytvožil virtuální property finvoice (f jako form) která je bool a setter se chová jako překladač té bool hodnoty na ten string 'yes' 'no' a výsledek ukládá do invoice. Proto je třeba změnit i název formulářového pole ve formuláři.

Reply:
a to si nemyslim ze je chtene. Do te prenosky jsme mohli dat cista data z formulare, ten nechat tak jednoduchy a citelny a stejne tak i tu prenosku. A pak v modelu co se stara o pridani do databaze si to poresit. Nemyslis ze by to tak bylo lepsi a citelnejsi? 

Jde tam pak i o otazku co s tim email, liame. Klidne bych to zrusil pokud to nejde osetrit v onValidate, ale nemam s tim zkusenost takze je mozne ze v onvalidate to zvladnout jeste jde. Pripadne pak doplnit neco jako https://contributte.org/packages/contributte/reCAPTCHA.html#usage

Reply:
**Do te prenosky jsme mohli dat cista data z formulare** Přenoskou se stala entita, do které také posíláš čistá data z formuláře. 
**ten nechat tak jednoduchy a citelny a stejne tak i tu prenosku.** Myslíš, že tím, že se přidalo f na zacatek u 2 form polí se zhoršila čitelnost?
**A pak v modelu co se stara o pridani do databaze si to poresit.** Entita už si to v sobě řeší úplně stejně, jako kdyby se použila přenoska navíc. Kdyby byla použita přenoska, tak by navíc oproti tomuto řešení musela být přidána metoda na mapování properties  'přenoska -> entita' stejně, jako jsi to měl dělané předtím přes konstruktor a tím pádem jediné čeho si dosáhl by byla úplně zbytečná třída navíc. 'setMappedType' metoda je pokud to správně chápu přidána proto, aby si žádnou přenosku nepotřeboval a zbavil se tak mezikroku a to mapování formuláře na entitu, jelikož ti to formulář udělá rovnou sám a ty nemusíš řešit žádnou logiku jen persistneš entitu. Ušetří to takto kódu hromady.
**Nemyslis ze by to tak bylo lepsi a citelnejsi?** Viz předchozí bod. Jak NF vyznává princip "less code, more security" přidávat přepravku navíc je kód navíc a rozhodně mi to nepřijde čitelnější, když kód, který jsem udělal by stejně musel existovat.

**Jde tam pak i o otazku co s tim email, liame.** '->getValues()' ti vrátí přímo instanci té třídy, která je specifikována v metodě 'setMappedType'. Takže tak, jak jsi to měl děláno to nepůjde z důvodů toho, že liame musí nastavit ten email. Proto jsem použil array access nad formulářem. Toto se ale dá vyřešit takto: '$form->getComponent('email')->getValue()' a to už projde. Takže bych to klidně jako ověření proti spamu nechal. :)

Reply:
Určitě ale, kdybych tu aplikaci psal já, tak bych do db ukládal rawData z formu tedy bool hodnotu a tím by tento problém celý byl vyřešen. Potom ten převod na yes a no by probíhal až při vykreslení. (Možná by to chtělo nějaký číselník v DB?)

Reply:
- **yes/no**  jelikoz to nema zadnou administraci tak k nahledu dat pouzivam adminer a je pro me prijemnejsi rovnou videt yes/no namisto 0/1. Ale stejne si to ted posilam emailem cely radek co si kopiruju do Google Sheets. Takze by se dal do budoucna upravit i DB navrh. 

- **prenoskou se stala Entita**, to je fajn, ale prijde mi ze se tam dela nejak moc divneho chovani v te entite proto aby se mohla stat i prenoskou pro formular. Jelikoz jsem to jeste nikde (tu prenosku) nepouzil tak s tim trosku v hlave bojuju 

- **liame / email** ... zkusim si nejaky test dneska at o tom dokazu napsat nejake relevantni informace


Imho tady narazim na takovej ten programatorskej pohled ze Entita = Formular a ja to moc nemam rad. Takhle vznikaji formulare a entity ktere nejsou uplne nejlepsi ani formular ani entita.

Reply:
Problém s f odstraněn.